wafer/users/views.py

```diff
@@ -87,7 +96,10 @@ class RegistrationView(EditOneselfMixin, FormView):
     template_name = 'wafer.users/registration.html'
 
     def get_user(self):
-        return UserProfile.objects.get(user__username=self.kwargs['username'])
+        try:
+            return UserProfile.objects.get(user__username=self.kwargs['username'])
+        except UserProfile.DoesNotExist:
+            raise Http404
```

Comment:
The point of this is not to give any reason. If we give a reason, we have to handle the privacy case specially.